src/Rfc6455/Message.php

```diff
@@ -133,4 +135,21 @@ public function isOperation()
     {
         return in_array($this->getFirstFrame()->getOpcode(), [Frame::OP_TEXT, Frame::OP_BINARY]);
     }
+
+    /**
+     * @throws ControlFrameException
+     * @throws TooBigFrameException
+     */
+    public function checkControlFrameSize()
+    {
+        $frame = $this->getFirstFrame();
+
+        if (!$frame->isControlFrame()) {
```

Comment:
The most important check is the fact that a control frame [cannot be fragmented](https://tools.ietf.org/html/rfc6455#section-5.5). This mean that `count($this->frames) === 1`.